Cython/Compiler/ExprNodes.py

```diff
@@ -2917,6 +2919,20 @@ def analyse_types(self, env):
                 IdentifierStringNode(self.module_name.pos, value=part)
                 for part in map(StringEncoding.EncodedString, self.module_name.value.split('.'))
             ]).analyse_types(env)
+        if self.level > 0:
+            qualname = env.global_scope().qualified_name
+            for _ in range(self.level):
+                qualname, _, _ = qualname.rpartition('.')
```

Comment:
I wonder if we should be prepared for this to run out of '.'s and not generate a valid qualname. In which case qualname should probably just be `NULL` and the optimized loopup skipped